### PR TITLE
Update to LLVM 18.1.0 rc 4

### PR DESCRIPTION
**Pull request by nikic**
This includes a fix to the libLLVM symlink needed to complete the update in https://github.com/rust-lang/rust/pull/121395.

r? @cuviper or @nagisa

### CODE REVIEW

**Comment by nagisa** [comment]
r=me

**Comment by nikic** [comment]
@nagisa On this repo, you have to click GitHub's button :)

**Review by nagisa — APPROVED** [review]

**Comment by nagisa** [comment]
Oh yeah, it enables the merge button then. You'll know better which merge strategy to use, so I'll leave it up to you to click it.

**Pull request merged**
